### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by TrisTenuto**

### CODE REVIEW

File: src/app/admintools/admintools.component.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        //Add ModeratorRights
+      }
+    }
+  }
+
+  toggleItem(ticket: ReportedItem) {
+    let index = this.tickets.indexOf(ticket);
+    this.collapsedItem[index] = !this.collapsedItem[index];
+  }
+}
```
Comment (clarson217) [inline src/app/admintools/admintools.component.ts]: Needs comments (probably)

File: src/app/discussion/discussion.component.ts [hunk 2]
```diff
@@ -89,4 +252,65 @@ export class DiscussionComponent implements OnInit {
...
+
+  getUserFollowedDis(){
+    this._forum.getUserFollowedDiscussion(this.userid).subscribe(data => {
+      data.forEach(dis => {
+        if(dis.discussionId == this.discussionID){
+          this.displayFollow = false;
+        }
+      });
+    })
+  }
```
Comment (clarson217) [inline src/app/discussion/discussion.component.ts:339]: Some methods need comments

File: src/app/error.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+}
+
+export class CustomError {
+  constructor(msg: string) {
+    this.errorMessage = msg;
+  }
+  errorMessage: string;
+}
+
```
Comment (clarson217) [inline src/app/error.service.ts:40]: Needs Comments

File: src/app/forum.service.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  postComment(newComment: any) {
+    return this.http.post(this.forumsUrl + "forum/comment", newComment);
+  }
+
+  submitDiscussion(discussion: any) {
+    return this.http.post(this.forumsUrl + "forum/discussion", discussion);
+  }
+}
```
Comment (clarson217) [inline src/app/forum.service.ts:90]: getDiscussion, getTopics, getUserDiscussions, getUserComments, getUserFollowedDiscussion, postComment, submitDiscussion  need comments

File: src/app/get-started-box/get-started-box.component.scss [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    margin-top: -3px;
+  }
+
+  //   svg {
+  //     display: inline-block;
+  //   }
+
+  //   input {
+  //     display: inline-block;
+  //   }
```
Comment (clarson217) [inline src/app/get-started-box/get-started-box.component.scss]: Should remove commented blocks I think.

File: src/app/get-started-box/get-started-box.component.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      ).length - 1);
+      window.location.href = "/list/" + searchParam + "/1";
+    }
+  }
+
+  authLogin() {
+    this.auth.login();
+  }
+
+}
```
Comment (clarson217) [inline src/app/get-started-box/get-started-box.component.ts:54]: Comments

File: src/app/home/home.component.ts [hunk 7]
```diff
@@ -1,59 +1,26 @@
...
-    this.logger.log("", this.currentUser);
-    return this.currentUser.username;
-  }
-  getName() {
-    this.logger.log("", this.currentUser.firstname);
-    return this.currentUser.firstname;
+    this.auth.authModel$.subscribe(reply => {
+      this.logger.log("current user in home", reply);
+      this.currentUser = reply;
+    });
```
Comment (clarson217) [inline src/app/home/home.component.ts:25]: Comment?

File: src/app/movie-discussions/movie-discussions.component.scss [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+// ul,
+// li {
+//   margin: 0;
+//   padding: 0;
+// }
+
+// ul {
+//   padding-left: 8%;
+//   margin-bottom: 5vh;
+// }
```
Comment (clarson217) [inline src/app/movie-discussions/movie-discussions.component.scss:39]: Should remove commented code

File: src/app/movie/movie.component.ts [hunk 9]
```diff
@@ -99,182 +100,66 @@ export class MovieComponent implements OnInit {
...
     setTimeout(() => {
-      this._login.getDiscussion(this.movieID).subscribe(data => {
-        this.logger.log("", data);
-        this.discussions = data;
-      });
+      // this.movieService.getMovieDiscussion(this.movieID).subscribe(data => {
+      //   console.log(data)
+      //   this.logger.log("", data);
+      //   this.discussions = data;
+      // });
```
Comment (clarson217) [inline src/app/movie/movie.component.ts]: Remove commented code

File: src/app/movie/movie.component.ts [hunk 10]
```diff
@@ -99,182 +100,66 @@ export class MovieComponent implements OnInit {
...
+  //     // this.logger.log("", this.user);
+  //     this.submitDiscussion.userid = this.userId;
+  //     this.sumbitReview.usernameid = this.username;
+  //     this.logger.log("", this.sumbitReview);
+
+  //   } else {
+
+  //     this.logger.log("", "no User");
+  //   }
+  // }
```
Comment (clarson217) [inline src/app/movie/movie.component.ts]: Remove commented code

File: src/app/profile/profile.component.ts [hunk 11]
```diff
@@ -110,33 +187,77 @@ export class ProfileComponent implements OnInit {
...
   editUser(): void {
     if (!this.userIsUpdating) {
       this.userIsEditable = true;
     }
   }
+
+  authLogout() {
+    this.auth.logout();
+  }
 }
```
Comment (clarson217) [inline src/app/profile/profile.component.ts:377]: Comments

File: src/app/review/review.component.ts [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+    // if (this.sumbitReview.score == 0 || this.sumbitReview.review == "") {
+    //   this.logger.log("", "Review Not Sumbitted");
+    // } else if (this.sumbitReview.review.length >= 250) {
+    //   alert("Reviews should be less than 250 Characters")
+    // } else {
+    //   this.reviewService.postMovieReview(this.sumbitReview).subscribe(data => this.logger.log("", data));
+    //   this.lastPage = false;
+    //   this.reloadReviews(true);
+    // }
+    // this.logger.log("", this.sumbitReview);
```
Comment (clarson217) [inline src/app/review/review.component.ts]: Remove commented code.

File: src/app/url.service.ts [hunk 13]
```diff
@@ -11,32 +11,11 @@ export class UrlService {
   MovieAPIUrl: string = "http://20.94.153.81/";
   AdminToolsAPIUrl: string = "http://20.189.29.112/";
   AuthenticationAPIUrl: string = "http://20.45.0.16/";
-  FrontendUrl: string = "http://20.94.137.143/";
+  // FrontendUrl: string = "http://20.94.137.143/";
```
Comment (clarson217) [inline src/app/url.service.ts]: Should this be taken out?

File: src/app/user.service.ts [hunk 14]
```diff
@@ -2,25 +2,133 @@ import { HttpClient, HttpClientModule } from '@angular/common/http';
...
+
+  getCurrentUser() {
+    this.logger.log("", this.currentUser);
+    return this.currentUser;
+  }
+
+  getAskingUser() {
+    this.logger.log("", this.askingUser);
+    return this.askingUser;
   }
```
Comment (clarson217) [inline src/app/user.service.ts:150]: Needs comments

**Review by clarson217 — CHANGES_REQUESTED** [review]
There's quite a few methods that need comments, and a few blocks of commented-out code  to be removed

**Review by clarson217 — APPROVED** [review]

**Review by markmoorerev — APPROVED** [review]

**Pull request merged**